packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -115,19 +123,45 @@ export function makeVatKeeper(
   }
 
   function setReachableFlag(kernelSlot) {
-    // const { type } = parseKernelSlot(kernelSlot);
+    const { type } = parseKernelSlot(kernelSlot);
     const kernelKey = `${vatID}.c.${kernelSlot}`;
-    const { vatSlot } = parseReachableAndVatSlot(kvStore.get(kernelKey));
-    // const { allocatedByVat } = parseVatSlot(vatSlot);
+    const { isReachable, vatSlot } = parseReachableAndVatSlot(
+      kvStore.get(kernelKey),
+    );
+    const { allocatedByVat } = parseVatSlot(vatSlot);
```

Comment:
Should there be something that combines `parseReachableAndVatSlot` and `parseVatSlot`?  It seems weird to parse something to get a thing which you then parse.  Not a deep concern, just wondering about hygiene.

Reply:
maybe.. they're nested, though. I'd have to look at the callers of `parseReachableAndVatSlot` to see how many of them also decompose the vref further, I'd bet it's like 50%